application/src/test/java/org/togetherjava/tjbot/commands/mediaonly/MediaOnlyChannelListenerTest.java

```diff
@@ -0,0 +1,91 @@
+package org.togetherjava.tjbot.commands.mediaonly;
+
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.entities.*;
+import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
+import net.dv8tion.jda.api.requests.RestAction;
+import net.dv8tion.jda.api.requests.restaction.AuditableRestAction;
+import net.dv8tion.jda.internal.JDAImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.togetherjava.tjbot.config.Config;
+
+import java.util.Collections;
+import java.util.List;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+class MediaOnlyChannelListenerTest {
+
+    private final JDA api = mock(JDA.class);
+
+    private final JDAImpl jda = mock(JDAImpl.class);
```

Comment:
You dont have to mock everything. we actually have `JdaTestHelper` or something like that (see the other unit tests). It comes with a lot of pre-done configuration and I bet it will simplify some stuff here.